src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs

```diff
@@ -15,7 +17,17 @@ namespace NJsonSchema.CodeGeneration
     public abstract class ValueGeneratorBase
     {
         private readonly CodeGeneratorSettingsBase _settings;
-
+        private readonly List<string> _fomatNotCompatibileWithString = new List<string>() {
+            JsonFormatStrings.Date,
+            JsonFormatStrings.DateTime,
+            JsonFormatStrings.Time,
+            JsonFormatStrings.TimeSpan,
+            JsonFormatStrings.Uri,
+            JsonFormatStrings.Guid,
```

Comment:
The exclusion of uri guid and uuid should only be applied to the csharp value generator and not to the typescript one...